core/src/main/java/org/apache/calcite/rex/RexCall.java

```diff
@@ -155,15 +237,15 @@ private static boolean equalSansNullability(RelDataType a, RelDataType b) {
         && x.startsWith(y);
   }
 
-  protected @Nonnull String computeDigest(boolean withType) {
+  protected @Nonnull String computeDigest(boolean withType, boolean withNormalize) {
     final StringBuilder sb = new StringBuilder(op.getName());
```

Comment:
I would suggest to introduce a new config class for computing digest, something like `computeDigest(RexCall.Config)` to give the interface better extensibility.

Reply:
Frankly speaking, I would refrain from extracting configuration classes and making this configurable unless there are clear use cases

Reply:
I've reverted `, boolean withNormalize` argument, and replaced it with `protected static boolean needNormalize()`.